### PR TITLE
B.O.S.S.: Bluespace Object Support System

### PR DESCRIPTION
**Pull request by Tacolizard**
:cl: Tacolizard: Bluespace Boogaloo
add: Added B.O.S.S., a new way for RnD to easily aid miners on lavaland by sending them gear. 
add: A B.O.S.S. sender unit has been added to science in Boxstation and Metastation. Insert any item with a tech origin and press 'Send all items' to instantly teleport them.
add: A B.O.S.S. receiver unit has been placed in the mining base. It receives items teleported to it from the sender located on-station.
/:cl:

before you ask, yes, bombs have a tech origin.

This PR is designed to facilitate a better relationship between miners and RnD by allowing scientists to instantaneously transport items to mining without even leaving their department. 

~~semi-wip, but pretty much conceptually complete. Just needs bugtesting and the like.~~
actually this is pretty much bug free yay!

**Sprites needed:**
B.O.S.S. sender
B.O.S.S. receiver

### CODE REVIEW

**Comment by Xhuis** [comment]
this feels like it might BOSS the players around!

**Comment by ChangelingRain** [comment]
Wow look it's like the paradox bag except you can throw anything with tech instead of anything that can fit and it's stationary

**Comment by Xhuis** [comment]
Well, technically, lavaland is just "wow look it's mining but now with flashy monsters and it's on fire"

Give it a chance! It has its own merits, too, and fills a different niche. Although if these QoL changes keep up, we might as well just replace shaft miners with a cardboard box named "DA MINERS" that generates ores every few minutes.

**Comment by Tacolizard** [comment]
A big part of this PR is to make lavaland bomb-mining more practical.

**Comment by kevinz000** [comment]
@ChangelingRain WOW ITS ALMOST AS IF HE'S ADDING A ROUNDSTART VERSION OF A DEVICE THAT EXISTS
@XDTM JUST HAPPENS TO HAVE CODED IT

IT STARTS WITH A Q AND ENDS WITH A PAD.

**Comment by Tacolizard** [comment]
the whole point of this is that if there's a big ol' machine in the middle of RnD that requires 0 effort to use or make, then mabye RnD will actually do their job and be nice and send the miners some equipment or bombs.

**Comment by kevinz000** [comment]
either
1. make them do their jobs and not be lazy fuckwits and build quantum pads (I've never seen miners ask for them!)
2. make them walk over
3. add quantum pads linked at roundstart to rnd and mining 

not a new item.

**Comment by Xhuis** [comment]
Good grief, don't lose your marbles about one change. He also has a right to disagree with their implementation and attempt to offer a solution that's less powerful in exchange for being available earlier.

**Comment by SgtxRootbeer** [comment]
these should work with mecha too, otherwise, its useless in my eyes

**Comment by Tacolizard** [comment]
@kevinz000 except quantum pads are balanced totally differently. 

this only allows items with a tech origin, and can only connect to a single receiver.

**Comment by Magikarpet** [comment]
disregarding that can you build the machine or is it a structure that dissapears after being broken

**Comment by kevinz000** [comment]
improve quantum pads to allow overridable checks and shit.
don't need a brand new machine when it already exists.

**Comment by Tacolizard** [comment]
@kevinz000 I'm not doing that. This isn't the same machine.

@Magikarpet it is incapable of being built for balance reasons. I am considering making it indestructible, however. Currently it is not.

**Comment by Repukan** [comment]
>Can't buff miners under the pretense that RnD has to wear their big boy pants and do their job
>We can however nerf module drops because miners aren't personally delivering ores to RnD and picking up ore is too much work for our poor scientists

Lmao!

**Comment by Tacolizard** [comment]
@Repukan #notallcoders

I'm not at all involved with those nerfs.

**Comment by Repukan** [comment]
I'm well aware, it just makes me laugh in this instance some are actually using "Make scientists work" as an excuse or argument when it wasn't accepted when mining nerfs were passed.

**Comment by Tacolizard** [comment]
@Repukan yeah I misinterpreted your comment as being against this PR. I totally agree with you.

**Comment by Ergovisavi** [comment]
I guess I don't get the point of this object.

Miners currently deliver ore to the station, then if they want science goodies, walk the ore over to R&D and ask for them. I guess this just lets miners drop off ore at the ORM and then go straight back to mining?

Like, if this is a quality of life thing, I don't think it's a significant QOL improvement, so I just question the purpose of the whole thing.

**Comment by Repukan** [comment]
I'm guessing it'll be more used in those situations where RnD is not up to levels to upgrade your equipment yet. So you'd drop off the mats and you'd get a message via radio that supply drops are being sent, but then you realize someone emmaged the fucking thing and now you're getting firebombed by ash lizards, Charlie we need air support the fucking ashcongs are blowing us up reeeeeeeeeeeeeeeeeeeeeeeeeee

**Comment by Tacolizard** [comment]
It lets miners go back to mining quicker. It also means people won't get arrested for carrying bombs through the station. Miners often don't bother going to science because a lot of the time they're ignored, and while they walk there they could be getting loot. If the equipment shows up right at their doorstep, then they'll feel more incentivized to mine. Likewise, with a giant machine in the middle of RnD, scientists will drop in some extra stuff to help the miners.

**Comment by RandomMarine** [comment]
This feels too QoL-creepy.

**Comment by optimumtact** [comment]
they have to get back to their pvp click simulator as fast as possible RandomMarine, doing actual job related work is bad and gets in the way of this

**Comment by RemieRichards** [comment]
This is bad.

File: code/game/machinery/boss.dm [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	..()
+	for(var/obj/machinery/boss_receiver/B in GLOB.boss_receivers)
+		if(B.is_mining_receiver)
+			linked = B
+			return
+
+/obj/machinery/boss_sender/attack_hand(mob/user)
+	..()
+	if(powered())
+		switch(alert(user,"Select an option.","[src.name] panel","Send all items","Manage inventory"))
```
Comment (RemieRichards) [inline code/game/machinery/boss.dm]: >alert

what is this 1995?

Comment (Tacolizard) [inline code/game/machinery/boss.dm]: i cant into tgui/i hate tgui and it really isn't necessary for this

**Comment by Grazyn** [comment]
>It lets miners go back to mining quicker

I thought the idea was that whatever keeps people on the station or at least makes them interact with others even a little is good for gameplay? It's also the reason why things like efficiency station which put the mining shuttle right in front of science never took off. The short wallk you have to take as a miner to reach science means that you're involved with whatever is happening on the station instead of interacting with the round only when it's cult and they set up a base on the outpost.

File: code/game/machinery/boss.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/machinery/boss.dm]: if you can't do tgui.
and you hate tgui.

do a ```/datum/browser``` UI you muppet.

**Comment by FantasticFwoosh** [comment]
If it means that miners won't aggressively crowd RnD standing around preening & pouting for their big boy toys they expect and demand like little children then im all for it. Let alone break in to be a dick & moonlight.

Set up the B.O.S.S, keep in radio contact, throw other non-tech supplies into a built teleporter to the mining base beacon (which really is the alternative)
________________________________
The only irk i find with this is, is it protected in the case golems or other people start making B.O.S.S and send shit to the mining base like free golems for a aggressive takeover? Or is that just emergent gameplay?

File: code/game/machinery/boss.dm (hunk 1 above)
Comment (lzimann) [inline code/game/machinery/boss.dm]: Alert/input on machinery as their UI is really bad. Do a browser datum like remie said.

**Comment by lzimann** [comment]
Honestly, I'm tempted to close this over the fact it's just ""QoL"". It might not be perfect, but forcing miners to go back is a valid way of them interacting with the station instead of being in their PVE land forever, so I will wait on @KorPhaeron to give his opinion on this.

**Comment by Cyberboss** [comment]
To add to everyone else's arguments we already have quantum pads. A maintainer who thinks this is worth can reopen it if they please

**Comment by Repukan** [comment]
Hey glad to see we never got Kor's opinion on this or that the 24 hour open thing didn't even apply.

At least now it's clear what kinda changes do go through

**Comment by Cyberboss** [comment]
If you really care that much, but I'm willing to bet Kor will agree

**Comment by Repukan** [comment]
Then let him agree

**Comment by Hatfish** [comment]
As someone said earlier, just build quantum pads if you want to send stuff to mining, you can even set them up to be one-way transmitters if you just want to send them stuff and not get a bunch of miners in your lab.  If miners can't be assed to walk over to R&D to get better equipment then that's their problem.

**Comment by Tacolizard** [comment]
Alright well overwhelming majority says no, and I don't feel like redoing the UI in a PR that will most likely not be merged.

**Pull request closed without merging**